### PR TITLE
Add function maker

### PR DESCRIPTION
**Pull request by blink1073**
This implements #63.  The only change to existing behaviour is to the interface to `run_func`.

### CODE REVIEW

**Comment by isbadawi** [comment]
Looks really cool! I'll let Ariel look at this before merging.

File: pymatbridge/pymatbridge.py [hunk 1]
```diff
@@ -433,3 +500,64 @@ def _preamble_code(self):
...
+# ----------------------------------------------------------------------------
+class Method(object):
+
+    def __init__(self, parent, name):
+        """An object representing a Matlab function
+
+        Methods are dynamically bound to instances of Matlab objects and
+        represent a callable function in the Matlab subprocess.
+
+        Args:
```
Comment (arokem) [inline pymatbridge/pymatbridge.py]: Please use the numpy docstring standard.

File: pymatbridge/pymatbridge.py [hunk 2]
```diff
@@ -433,3 +500,64 @@ def _preamble_code(self):
 
     def _execute_flag(self):
         return '--eval'
+
+
+# ----------------------------------------------------------------------------
+# MATLAB METHOD
+# ----------------------------------------------------------------------------
+class Method(object):
```
Comment (arokem) [inline pymatbridge/pymatbridge.py]: `Method` is a very confusing name for an object

Comment (blink1073) [inline pymatbridge/pymatbridge.py]: MatlabMethod?

**Comment by blink1073** [comment]
Note that a bulk of the code was pulled from #63.

File: pymatbridge/pymatbridge.py (hunk 2 above)
Comment (arokem) [inline pymatbridge/pymatbridge.py]: Perhaps MatlabFunction? Do you bind anything but functions with this?

Comment (blink1073) [inline pymatbridge/pymatbridge.py]: Technically the name just has to exist, but MatlabFunction works for me.

Comment (arokem) [inline pymatbridge/pymatbridge.py]: Yeah - let's go with that. Other than that - looks good to me. Let me give it another read tomorrow, before going ahead with this. Thanks a lot for taking this on! That split in the code-base has been bothering me for a long while.

Comment (blink1073) [inline pymatbridge/pymatbridge.py]: Done, also added a check to make sure it is a function-like object.

**Comment by blink1073** [comment]
Open question: should `run_func` be made to look more like `run_code`, returning stdout and any created figures?  Also, should the dynamic functions return just the response or the full dictionary?  For example `m.plot([1,2,3])` would make the plot but we'd never see it.  If we got the path to the created figure, we could do something with it.

**Comment by blink1073** [comment]
Here's what I propose:
Use `pymat_eval` for both, and either run `evalin` or a version of `pymat_feval` [here](https://github.com/arokem/python-matlab-bridge/blob/master/pymatbridge/matlab/util/pymat_eval.m#L38).

**Comment by arokem** [comment]
Just to see that I understand: this would grab the figures and stdout in
the magic even if you call a function that creates the figure? I like that.

**Comment by blink1073** [comment]
To be clear, the matlab magic already uses `run_code` exclusively, so this would affect `run_func` and the dynamically created functions.

**Comment by blink1073** [comment]
I think this is good to go now.  I will make the changes to the Matlab end in another PR.

**Comment by arokem** [comment]
Gotcha. Looks good!

**Pull request merged**